app/api/search/search.js

```diff
@@ -147,96 +161,193 @@ const processResponse = (response) => {
+          result[value.id] = translate(context, value.label, value.label);
+          return result;
+        }, {});
+        documentsQuery.sortByForeignKey(query.sort, keys, query.order);
+      } else {
+        documentsQuery.sort(query.sort, query.order);
       }
+    }
 
-      const allTemplates = templates.map(t => t._id);
-      const allUniqueProps = allUniqueProperties(templates);
-      const filteringTypes = query.types && query.types.length ? query.types : allTemplates;
-      let properties = comonFilters(templates, relationTypes, filteringTypes);
-      properties = !query.types || !query.types.length ? defaultFilters(templates) : properties;
-
-      if (query.sort) {
-        const sortingProp = allUniqueProps.find(p => `metadata.${p.name}` === query.sort);
-        if (sortingProp && sortingProp.type === 'select') {
-          const dictionary = dictionaries.find(d => d._id.toString() === sortingProp.content);
-          const translation = getLocaleTranslation(_translations, language);
-          const context = getContext(translation, dictionary._id.toString());
-          const keys = dictionary.values.reduce((result, value) => {
-            result[value.id] = translate(context, value.label, value.label);
-            return result;
-          }, {});
-          documentsQuery.sortByForeignKey(query.sort, keys, query.order);
-        } else {
-          documentsQuery.sort(query.sort, query.order);
-        }
-      }
+    if (query.allAggregations) {
+      properties = allUniqueProps;
+    }
 
-      if (query.allAggregations) {
-        properties = allUniqueProps;
-      }
+    const aggregations = agregationProperties(properties);
+    const filters = processFiltes(query.filters, properties);
+    const textSearchFilters = filtersBasedOnSearchTerm(allUniqueProps, entitiesMatchedByTitle, dictionariesMatchByLabel);
 
-      const aggregations = agregationProperties(properties);
-      const filters = processFiltes(query.filters, properties);
-      const textSearchFilters = filtersBasedOnSearchTerm(allUniqueProps, entitiesMatchedByTitle, dictionariesMatchByLabel);
 
+    documentsQuery.filterMetadataByFullText(textSearchFilters);
+    documentsQuery.filterMetadata(filters);
+    documentsQuery.aggregations(aggregations, dictionaries);
 
-      documentsQuery.filterMetadataByFullText(textSearchFilters);
-      documentsQuery.filterMetadata(filters);
-      documentsQuery.aggregations(aggregations, dictionaries);
+    if (query.geolocation) {
+      searchGeolocation(documentsQuery, filteringTypes, templates);
+    }
+    return elastic.search({ index: elasticIndex, body: documentsQuery.query() })
+    .then(processResponse)
+    .catch(() => {
+      throw createError('Query error', 400);
+    });
+  });
+};
 
-      if (query.geolocation) {
-        searchGeolocation(documentsQuery, filteringTypes, templates);
+const determineInheritedProperties = templates => templates.reduce((memo, template) => {
```

Comment:
why not just call the argument memo inheritedProperties?
```suggestion
const determineInheritedProperties = templates => templates.reduce((inheritedProperties , template) => {

```